// @seniorsistemas/fsw-aws-lambda/lib/g5Api/index.js
const axiosW = require('../utils/axios-wrapper');

class g5Api {

    constructor(sxiServer,sxiContext,g5Server,user,token) {
        this._sxiServer = sxiServer;
        this._sxiContext = sxiContext;
        this._g5Server = g5Server;
        this._url = `${this._sxiServer }/${this._sxiContext}`;
        this._user = user; 
        this._token = token;
    }

    _config(user, token,g5Module,service,port,dataSourceName) {

        let config = {};
    
        const headers = {
            'Content-Type': 'application/json' ,
            user: user
        };

        if(token.indexOf('Bearer') !== -1) {
            headers.Authorization = token;
        } else {
            headers.pass = token;
            headers.encryptionType = 0;
        }

        const params = {
            server: this._g5Server,
            module: g5Module,
            service: service,
            port: port
        };

        if(dataSourceName){
            params.DATASOURCEATTRNAME = dataSourceName;
        }

        config = { headers, params };

        return config;
    }

    async callService  (g5Module,service,port,body,dataSourceName)  {
        return await axiosW.post(`${this._url}/G5Rest`, body, this._config(this._user,this._token,g5Module,service, port, dataSourceName));
    }
}

module.exports = g5Api;
