// @seniorsistemas/fsw-aws-lambda/lib/services/email-service.js
const axiosW = require('./../utils/axios-wrapper');

async function sendEmail (event,to,subject,content) {
    const destinators = to.split(',');

    return axiosW.platformPost(event,'/platform/notifications/actions/notifyUser', {
        notificationClass : 'TPS_REPORT_VALIDATION',
        notificationOrigin : 'Customization',
        notificationKind : 'News',
        notificationPriority : 'None',
        notificationSubject : subject,
        notificationContent : content,
        sourceDomain : 'platform',
        sourceService : 'notifications',
        to:  destinators,
        format: 'PLAIN_TEXT',
        _discriminator: 'emailNotificationPayload'
    });
}

async function sendEmailHTML(event,to,subject,content) {
    const destinators = to.split(',');
    
    return axiosW.platformPost(event,'/platform/notifications/actions/notifyUser', {
        notificationClass : 'TPS_REPORT_VALIDATION',
        notificationOrigin : 'Customization',
        notificationKind : 'News',
        notificationPriority : 'None',
        notificationSubject : subject,
        notificationContent : content,
        sourceDomain : 'platform',
        sourceService : 'notifications',
        to:  destinators,
        format: 'HTML',
        _discriminator: 'emailNotificationPayload'
    });
}


module.exports = {
    sendEmail,
    sendEmailHTML
};
